supplementary/alica_ros1/alica_tests/src/test/test_blackboard.cpp

```diff
@@ -117,5 +129,638 @@ TEST_F(TestBlackBoard, testMapping)
+}
+
+TEST_F(TestBlackboard, testSetNotMatchingKnownType)
+{
+    std::unique_ptr<alica::BlackboardBlueprint> blueprint = std::make_unique<alica::BlackboardBlueprint>();
+    blueprint->addKey("intVal", "int64", "14");
+    blueprint->addKey("doubleVal", "double", "2.7");
+    blueprint->addKey("boolVal", "bool", "true");
+    blueprint->addKey("stringVal", "std::string", "test string");
+    alica::Blackboard blackboard = alica::Blackboard(blueprint.get());
+    alica::LockedBlackboardRW bb = LockedBlackboardRW(blackboard);
+
+    EXPECT_EQ(bb.get<int64_t>("intVal"), 14);
+    EXPECT_EQ(bb.get<double>("doubleVal"), 2.7);
+    EXPECT_EQ(bb.get<bool>("boolVal"), true);
+    EXPECT_EQ(bb.get<std::string>("stringVal"), "test string");
+
+    EXPECT_THROW(
+            {
+                try {
+                    bb.set<double>("intVal", 2.7);
+                } catch (const BlackboardException& e) {
+                    EXPECT_STREQ(e.what(), "Blackboard exception: set() type mismatch, key: intVal, set type: double, expected type (from yaml file): int64");
+                    throw;
+                }
+            },
+            BlackboardException);
+
+    EXPECT_THROW(
+            {
+                try {
+                    bb.set<std::string>("doubleVal", "test");
+                } catch (const BlackboardException& e) {
+                    EXPECT_STREQ(e.what(),
+                            "Blackboard exception: set() type mismatch, key: doubleVal, set type: std::string, expected type (from yaml file): double");
+                    throw;
+                }
+            },
+            BlackboardException);
+
+    EXPECT_THROW(
+            {
+                try {
+                    bb.set<std::string>("boolVal", "test");
+                } catch (const BlackboardException& e) {
+                    EXPECT_STREQ(
+                            e.what(), "Blackboard exception: set() type mismatch, key: boolVal, set type: std::string, expected type (from yaml file): bool");
+                    throw;
+                }
+            },
+            BlackboardException);
+
+    EXPECT_THROW(
+            {
+                try {
+                    bb.set<int64_t>("stringVal", 0);
+                } catch (const BlackboardException& e) {
+                    EXPECT_STREQ(e.what(),
+                            "Blackboard exception: set() type mismatch, key: stringVal, set type: int64, expected type (from yaml file): std::string");
+                    throw;
+                }
+            },
+            BlackboardException);
+}
+
+TEST_F(TestBlackboard, testSetNotMatchingUnknownType)
+{
+    Blackboard bb;
+    auto bb_locked = LockedBlackboardRW(bb);
+    bb_locked.set<PlanStatus>("value", PlanStatus::Success);
+    EXPECT_EQ(bb_locked.get<PlanStatus>("value"), PlanStatus::Success);
+
+    bb_locked.set<UnknownType>("value", UnknownType());
+    EXPECT_EQ(typeid(UnknownType), typeid(bb_locked.get<UnknownType>("value")));
```

Comment:
types are definitely going to match because get returns the type that it was called with. What needs to be checked is if the values match. You can make unknown type a wrapper for an int i.e. add a constructor which takes an int & check if the right value is returned